### PR TITLE
[WIP] Fix annotation get window extent

### PR DESCRIPTION
**Pull request by itziakos**
This PR  reworks `Annotation` to provide a get_window_extent method that will return as bounding box the union of the contained text and arrow.

Pending:
- [ ] more tests 
- [ ] cleanup

### CODE REVIEW

**Comment by tacaswell** [comment]
Great!

This looks like it will need a re-base to current master already.

**Comment by itziakos** [comment]
I am on it

**Comment by itziakos** [comment]
conflict resolved, please note that I still need to fix and add a couple of tests

File: lib/matplotlib/tests/test_text.py [hunk 1]
```diff
@@ -279,8 +279,127 @@ def test_get_rotation_mod360():
...
 
+
 @image_comparison(baseline_images=['text_bboxclip'])
 def test_bbox_clipping():
     plt.text(0.9, 0.2, 'Is bbox clipped?', backgroundcolor='r', clip_on=True)
     t = plt.text(0.9, 0.5, 'Is fancy bbox clipped?', clip_on=True)
     t.set_bbox({"boxstyle": "round, pad=0.1"})
+
+
+def test_text_annotation_get_window_extent():
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_text.py]: It is easier to add the `@cleanup` decorator use pyplot to make your figure/axes objects.

File: lib/matplotlib/tests/test_text.py [hunk 2]
```diff
@@ -279,8 +279,127 @@ def test_get_rotation_mod360():
...
+    from matplotlib.figure import Figure
+    from matplotlib.text import Annotation, Text
+    from matplotlib.backends.backend_agg import RendererAgg
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_text.py]: Please move this to the top, no reason to do function-level imports here.

File: lib/matplotlib/tests/test_text.py (hunk 1 above)
Comment (tacaswell) [inline lib/matplotlib/tests/test_text.py]: Maybe, I now see why you are doing it this way.

**Comment by tacaswell** [comment]
I have cherry-picked these over to be based on the 1.4.x branch.  I should have a version of this merged in the near future.

**Comment by tacaswell** [comment]
Merged in to v1.4.x as 26c7d656de106534b1871433c965dac9789ef264

**Comment by itziakos** [comment]
Thanks @tacaswell.

Sorry for the delay. I will make a separate PR for the additional tests.

**Comment by tacaswell** [comment]
@itziakos  Thanks for your work.  

Sorry if you feel like I stepped on your toes, I wanted to get this in and this was the fastest path to get there.

**Comment by itziakos** [comment]
> Sorry if you feel like I stepped on your toes, I wanted to get this in and this was the fastest path to get there.

There was no offense taken, It is me who should apologize for the delay.

**Pull request closed without merging**